packages/purpose-process/src/services/purposeService.ts

```diff
@@ -596,6 +607,133 @@ export function purposeServiceBuilder(
         totalCount: purposesList.totalCount,
       };
     },
+    async createPurposeVersion({
+      purposeId,
+      seed,
+      organizationId,
+      correlationId,
+      logger,
+    }: {
+      purposeId: PurposeId;
+      seed: ApiPurposeVersionSeed;
+      organizationId: TenantId;
+      correlationId: string;
+      logger: Logger;
+    }): Promise<PurposeVersion> {
+      logger.info(`Creating Version for Purpose ${purposeId}`);
+
+      const purpose = await retrievePurpose(purposeId, readModelService);
+
+      assertOrganizationIsAConsumer(organizationId, purpose.data.consumerId);
+      assertDailyCallsIsDifferentThanBefore(purpose.data, seed.dailyCalls);
+
+      const conflictVersion = purpose.data.versions.find(
+        (v) =>
+          v.state === purposeVersionState.draft ||
+          v.state === purposeVersionState.waitingForApproval
+      );
+
+      if (conflictVersion !== undefined) {
```

Comment:
```suggestion
      const isConflictVersionPresent = purpose.data.versions.some(
        (v) =>
          v.state === purposeVersionState.draft ||
          v.state === purposeVersionState.waitingForApproval
      );

      if (isConflictVersionPresent) {
```
Could we use `some` here instead of `find` in order to have a boolean? Feel free to rename the const

Reply:
Yup, good idea!

Reply:
Good idea!

Reply:
Actually no, it is not possible. I need the version that goes in conflict for error logging reason